configure.ac

```diff
@@ -74,6 +74,24 @@ AS_CASE("$derivative_type",
 
 AC_MSG_RESULT([configuring with derivative type $derivative_type])
 
+AC_ARG_WITH(libtorch,
+            AS_HELP_STRING([--with-libtorch@<:@=ARG@:>@],[Specify if libtorch support should be enabled @<:@ARG=yes or libtorch directory@:>@]),
+            [libtorch_support=yes libtorch_library="$withval"],
+            [libtorch_support=no])
+
+AC_MSG_RESULT([configuring with libtorch support: $libtorch_support])
+
+AS_IF([test "$libtorch_support" = yes],
+      [
+        AC_DEFINE(LIBTORCH_ENABLED, 1, [Whether to use libtorch-related code or not])
+        AC_SUBST([libtorch_support],[true])
+        AS_IF([test "$libtorch_library" = yes],
+              [libtorch_library="$ac_pwd/framework/contrib/libtorch"])
```

Comment:
Moose isn't anywhere near supporting out-of-source build directories, right?  If I'm wrong then we may need to change `$ac_pwd`, which would give the build dir rather than the source dir.

Reply:
Oh no, we don't use that for MOOSE and I don't see that being an option in the near future either.